lib/unit/api_resources/check_deposit_resource.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require_relative "./base_resource"
+require_relative "../utils/http_helper"
+require "json"
+# class for creating requests for check deposits to Unit API and parsing responses
+# @see https://docs.unit.co/check-deposits
+module Unit
+  module Resource
+    class CheckDepositResource < Unit::Resource::BaseResource
+      class << self
+        # Create a check deposit by calling Unit's API
+        # @param request [CreateCheckDepositRequest]
+        # @return [UnitResponse, UnitError]
+        def create_deposit(request)
+          payload = request.to_json_api
+          response = HttpHelper.post("#{api_url}/check-deposits", body: payload, headers: headers)
+          response_handler(response)
+        end
+
+        # Get a check deposit by id by calling Unit's API
+        # @param params [GetRequest]
+        # @return [UnitResponse, UnitError]
+        def get(params)
+          payload = params.to_hash
+          response = HttpHelper.get("#{api_url}/check-deposits/#{params.deposit_id}", params: payload, headers: headers)
+          response_handler(response)
+        end
+
+        # Get a check deposits by calling Unit's API
+        # @param params [ListDepositRequest]
+        # @return [UnitResponse, UnitError]
+        def list(params = nil)
+          response = HttpHelper.get("#{api_url}/check-deposits", params: params.to_hash, headers: headers)
+          response_handler(response)
+        end
+
+        # Update a check deposit by id by calling Unit's API
+        # @param request [PatchCheckDepositRequest]
+        # @return [UnitResponse, UnitError]
+        def update(request)
+          payload = request.to_json_api
+          response = HttpHelper.patch("#{api_url}/check-deposits/#{request.deposit_id}", body: payload, headers: headers)
+          response_handler(response)
+        end
+
+        # Upload a check deposit image by calling Unit's API
+        # @param request [ImageRequest]
+        # @return [UnitResponse, UnitError]
+        def upload(request)
+          url = "#{api_url}/check-deposits/#{request.deposit_id}"
+          url += "/front" if request.is_front_side
+          url += "/back" unless request.is_front_side
```

Comment:
use - https://www.rubyguides.com/2019/10/ruby-ternary-operator/
```suggestion
          url = "#{api_url}/check-deposits/#{request.deposit_id}" + (request.is_front_side ?  "/front" : "/back")
```
double check this.